openedx/features/course_experience/tests/views/test_course_outline.py

```diff
@@ -350,7 +355,65 @@ def test_resume_course(self):
         self.assertContains(response, 'Resume Course', count=2)
 
         content = pq(response.content)
-        self.assertTrue(content('.action-resume-course').attr('href').endswith('/sequential/' + sequential.url_name))
+        self.assertTrue(content('.action-resume-course').attr('href').endswith('/vertical/' + vertical.url_name))
+
+    def test_resume_course_with_completion_api(self):
+        """
+        Tests completion API resume button functionality
+        """
+        # Enable the waffle flag for all tests
+        self.override_waffle_switch(True)
+
+        # Course, Verticals
+        course = self.course
+        course_key = CourseKey.from_string(str(course.id))
+        vertical1 = course.children[0].children[0].children[0]
+        vertical2 = course.children[0].children[1].children[0]
+
+        # Fake a visit to sequence1/vertical1
+        block_key = UsageKey.from_string(unicode(vertical1.location))
+        completion = 1.0
+        BlockCompletion.objects.submit_completion(
+            user=self.user,
+            course_key=course_key,
+            block_key=block_key,
+            completion=completion
+        )
+        # Test for 'resume' link
+        response = self.client.get(course_home_url(course))
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, 'Start Course', count=0)
+        self.assertContains(response, 'Resume Course', count=2)
+
+        # Test for 'resume' link URL - should be vertical 1
+        content = pq(response.content)
+        self.assertTrue(content('.action-resume-course').attr('href').endswith('/vertical/' + vertical1.url_name))
+
+        # Fake a visit to sequence2/vertical2
+        block_key = UsageKey.from_string(unicode(vertical2.location))
+        completion = 1.0
+        BlockCompletion.objects.submit_completion(
+            user=self.user,
+            course_key=course_key,
+            block_key=block_key,
+            completion=completion
+        )
+        response = self.client.get(course_home_url(course))
+        self.assertEqual(response.status_code, 200)
+
+        # Test for 'resume' link URL - should be vertical 2
+        content = pq(response.content)
+        self.assertTrue(content('.action-resume-course').attr('href').endswith('/vertical/' + vertical2.url_name))
+
+        # visit sequential 1, make sure 'Resume Course' URL is robust against 'Last Visited'
+        # (even though I visited seq1/vert1, 'Resume Course' still points to seq2/vert2)
+        self.visit_sequential(course, course.children[0], course.children[0].children[0])
+        response = self.client.get(course_home_url(course))
```

Comment:
#NIT, the following 4 lines can be written in a helper function to keep the test dryer?

Reply:
@schenedx Good call! Done.